sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -1425,5 +1433,80 @@ Duration getDefaultPollingInterval() {
             return monoError(logger, e);
         }
     }
+
+    /**
+     * Releases a key.
+     *
+     * <p>The key must be exportable. This operation requires the 'keys/release' permission.</p>
+     *
+     * @param name The name of the key to release.
+     * @param target The attestation assertion for the target of the key release.
+     *
+     * @return A {@link Mono} containing the {@link ReleaseKeyResult} containing the released key.
+     *
+     * @throws IllegalArgumentException If {@code name} or {@code target} are {@code null} or empty.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ReleaseKeyResult> releaseKey(String name, String target) {
+        try {
+            return releaseKeyWithResponse(name, null, target, new ReleaseKeyOptions())
+                .flatMap(FluxUtil::toMono);
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * Releases a key.
+     *
+     * <p>The key must be exportable. This operation requires the 'keys/release' permission.</p>
+     *
+     * @param name The name of the key to release.
+     * @param version Version of the key to release.This parameter is optional.
+     * @param target The attestation assertion for the target of the key release.
+     * @param options Additional options for releasing a key.
+     *
+     * @return A {@link Mono} containing the {@link Response HTTP response} for this operation and the
+     * {@link ReleaseKeyResult} containing the released key.
+     *
+     * @throws IllegalArgumentException If {@code name} or {@code target} are {@code null} or empty.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ReleaseKeyResult>> releaseKeyWithResponse(String name, String version, String target,
```

Comment:
You are definitely right, it looks like when I originally wrote this I thought `version` was not required, but looking at the [service specification](https://github.com/Azure/azure-rest-api-specs/blob/301de051cd0e618222e7c1bb58c2a430828b9374/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json#L929) that is not the case.

Reply:
I will remove the `releaseKey(String name, String target)` method in favor of `releaseKey(String name, String version, String target)` as the minimum overload.

Reply:
I see .NET [has both overloads](https://github.com/Azure/azure-sdk-for-net/blob/49f841e464b688b1acf795ca32cd88f9ca8557d8/sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs#L140), but is it really allowed to send a release request without providing a version?

Reply:
A release request without a `version` will release the most recent key version by default, but you'll probably get an error if you send a null type instead of something like an empty string: https://github.com/Azure/azure-sdk-for-python/pull/20048#discussion_r682193171